seqr/views/utils/file_utils.py

```diff
@@ -86,25 +91,32 @@ def save_uploaded_file(request, process_records=None):
         raise ValueError("Received %s files instead of 1" % len(request.FILES))
 
     # parse file
-    stream = request.FILES.values()[0]
+    stream = next(iter(request.FILES.values()))
     filename = stream._name
 
+    if filename.endswith('.csv'):
+        try:
+            a = unicode
+        except:
+            stream = TextIOWrapper(stream.file, encoding = 'utf-8')
+    elif not filename.endswith('.xls') and not filename.endswith('.xlsx'):
+        stream = TextIOWrapper(stream.file, encoding = 'utf-8')
     json_records = parse_file(filename, stream)
     if process_records:
         json_records = process_records(json_records, filename=filename)
 
     # save json to temporary file
-    uploaded_file_id = hashlib.md5(str(json_records)).hexdigest()
+    uploaded_file_id = hashlib.md5(str(json_records).encode('utf-8')).hexdigest()
     serialized_file_path = _compute_serialized_file_path(uploaded_file_id)
-    with gzip.open(serialized_file_path, "w") as f:
+    with gzip.open(serialized_file_path, "wt") as f:
```

Comment:
The change from `w` to `wt` is required by Python 3 otherwise it can only write binary data while json.dump() generates text data.